assets/js/base/components/cart-checkout/address-form/address-form.tsx

```diff
@@ -137,6 +137,13 @@ const AddressForm = ( {
 		<div id={ id } className="wc-block-components-address-form">
 			{ addressFormFields.map( ( field ) => {
 				if ( field.hidden ) {
+					// clear hidden field value.
+					if ( values[ field.key ] ) {
+						onChange( {
+							...values,
+							[ field.key ]: '',
+						} );
+					}
```

Comment:
That means `onChange()` is called on each render, right? Could this be refactored so it's run inside a `useEffect()` when the country or state change?